libraries/common/stringbuffer.effekt

```diff
@@ -19,7 +19,8 @@ def stringBuffer[A] { prog: => A / StringBuffer }: A = {
     else {
       // Double the capacity while ensuring the required capacity
       val newSize = max(buffer.size * 2, buffer.size + sizeToAdd)
-      buffer = bytearray::resize(buffer, newSize)
+      val newBuffer = bytearray::allocate(newSize)
+      buffer = buffer.resize(newSize)
```

Comment:
Similarly here: `resize` already allocates, right?
https://github.com/effekt-lang/effekt/blob/26c642df51bcdbb22186e16069d7e523d4717a85/libraries/common/bytearray.effekt#L40-L41

... also, you never use `newBuffer` anyway? :^)